lib/spack/spack/cmd/develop.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+import spack.environment as ev
+
+from spack.error import SpackError
+
+description = 'add a spec to an environments dev-build information'
+section = "environments"
+level = "long"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-p', '--path', help='Source location of package')
+    subparser.add_argument(
+        '--no-clone', action='store_true',
+        help='Do not clone. The package already exists at the source path')
```

Comment:
Yes, this is so the user can specify their own source.

`Package.do_install` always sets `keep-stage=True` in part because of this, and in part because even if the user used Spack to clone the repo, they're going to want to keep that repo to commit the changes after they make sure it worked.

Reply:
I'm wondering if it would make sense to check the path to make sure it's empty. My first inclination is that if the path doesn't exist or is empty, we could just automatically clone. If anything is in the path, we would skip. Is there a reason you want to clone by default?